### PR TITLE
Prompt ensemble

### PR DESCRIPTION
**Pull request by jwehrmann**

### CODE REVIEW

**Comment by jwehrmann** [comment]
FYI @cardoso-neto

**Comment by cardoso-neto** [comment]
I'm going to need some context.

**Comment by jwehrmann** [comment]
Just look at the examples :sweat_smile:

**Comment by cardoso-neto** [comment]
Am I correct in assuming this would never be used in production, but just for testing out prompt ideas?

File: prompts/ensemble.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    def __init__(
+        self, templates: list[str] | Type[BasePrompt],
+        expected_vars: Optional[list[str]] = None,
+        prompt_class=DynamicPrompt,
+    ):
+        """
+        Args:
+            templates: templates with placeholder variable names
+            expected_vars: variables expected in all templates
+
```
Comment (martinduartemore) [inline prompts/ensemble.py]: The `prompt_class` argument is not documented

File: setup.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def get_optional_requirements() -> dict[str, list[str]]:
     url="https://github.com/TeiaLabs/Recruit",
     packages=setuptools.find_packages(),
     keywords="prompt openai teialabs gpt3",
-    python_requires=">=3.9",
+    python_requires=">=3.7",
```
Comment (martinduartemore) [inline setup.py:43]: :sob:

**Review by martinduartemore — COMMENTED** [review]
Everything seems great! :+1:

File: setup.py (hunk 2 above)
Comment (jwehrmann) [reply setup.py:43]: I don't like living by the edge of the knife

**Pull request merged**